From 441dfe759977cdb07c5b113b0d1ce945b4ac4cf1 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibochen@cloudera.com>
Date: Wed, 26 Apr 2017 11:43:27 -0700
Subject: [PATCH 2391/2748] YARN-6510. Fix profs stat file warning caused by
 process names that includes parenthesis. (Wilfred
 Spiegelenburg via Haibo Chen)

(cherry picked from commit 4f3ca0396a810f54f7fd0489a224c1bb13143aa4)

Change-Id: I1f08542b909cef233a75d0d8c547e8fd4b504300
---
 .../hadoop/yarn/util/ProcfsBasedProcessTree.java   |    2 +-
 .../yarn/util/TestProcfsBasedProcessTree.java      |    8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ProcfsBasedProcessTree.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ProcfsBasedProcessTree.java
index 75e0f38..ceace75 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ProcfsBasedProcessTree.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ProcfsBasedProcessTree.java
@@ -56,7 +56,7 @@
   private static final String PROCFS = "/proc/";
 
   private static final Pattern PROCFS_STAT_FILE_FORMAT = Pattern.compile(
-      "^([\\d-]+)\\s\\(([^)]+)\\)\\s[^\\s]\\s([\\d-]+)\\s([\\d-]+)\\s" +
+      "^([\\d-]+)\\s\\((.*)\\)\\s[^\\s]\\s([\\d-]+)\\s([\\d-]+)\\s" +
       "([\\d-]+)\\s([\\d-]+\\s){7}(\\d+)\\s(\\d+)\\s([\\d-]+\\s){7}(\\d+)\\s" +
       "(\\d+)(\\s[\\d-]+){15}");
 
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/util/TestProcfsBasedProcessTree.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/util/TestProcfsBasedProcessTree.java
index 9b3320c..f8f47a9 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/util/TestProcfsBasedProcessTree.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/util/TestProcfsBasedProcessTree.java
@@ -414,7 +414,7 @@ public void testCpuAndMemoryForProcessTree() throws IOException {
           new ProcessStatInfo(new String[]{"200", "process two", "100", "100",
               "100", "200000", "200", "2000", "400"});
       procInfos[2] =
-          new ProcessStatInfo(new String[]{"300", "proc3", "200", "100",
+          new ProcessStatInfo(new String[]{"300", "proc(3)", "200", "100",
               "100", "300000", "300", "3000", "600"});
       procInfos[3] =
           new ProcessStatInfo(new String[]{"400", "proc4", "1", "400", "400",
@@ -558,7 +558,7 @@ private void testMemForOlderProcesses(boolean smapEnabled) throws IOException {
           new ProcessStatInfo(new String[]{"200", "process two", "100", "100",
               "100", "200000", "200"});
       procInfos[2] =
-          new ProcessStatInfo(new String[]{"300", "proc3", "1", "300", "300",
+          new ProcessStatInfo(new String[]{"300", "proc(3)", "1", "300", "300",
               "300000", "300"});
       procInfos[3] =
           new ProcessStatInfo(new String[]{"400", "proc4", "100", "100",
@@ -810,7 +810,7 @@ public void testProcessTreeDump() throws IOException {
           new ProcessStatInfo(new String[]{"200", "process two", "100", "100",
               "100", "200000", "200", "2000", "400"});
       procInfos[2] =
-          new ProcessStatInfo(new String[]{"300", "proc3", "200", "100",
+          new ProcessStatInfo(new String[]{"300", "proc(3)", "200", "100",
               "100", "300000", "300", "3000", "600"});
       procInfos[3] =
           new ProcessStatInfo(new String[]{"400", "proc4", "200", "100",
@@ -833,7 +833,7 @@ public void testProcessTreeDump() throws IOException {
       String[] cmdLines = new String[numProcesses];
       cmdLines[0] = "proc1 arg1 arg2";
       cmdLines[1] = "process two arg3 arg4";
-      cmdLines[2] = "proc3 arg5 arg6";
+      cmdLines[2] = "proc(3) arg5 arg6";
       cmdLines[3] = "proc4 arg7 arg8";
       cmdLines[4] = "proc5 arg9 arg10";
       cmdLines[5] = "proc6 arg11 arg12";
-- 
1.7.9.5

